settings: Do not use list_render for stream specific notification settings and fix scrolling bugs.

sahil839 opened the pull request:
We do not need to use list_render for displaying list of streams specific
notification settings, as this is not scrollable and we do not provide
option to sort or filter this list as well.

After this change, all our list_render instances will be using simplebar
and we can change the code accordingly to fix the behaviour of scrolling.

<!-- What's this PR for?  (Just a link to an issue is fine.) -->


**Testing Plan:** <!-- How have you tested? -->


**GIFs or Screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->


<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

timabbott commented on static/js/settings_notifications.js:
```diff
@@ -47,17 +47,18 @@ function rerender_ui() {
-    });
+            }),
+        );
+    }
```
Have you tested this?  It certainly looks like we might be using the `modified` feature for some sort of live-update functionality.

sahil839 replied:
Yeah I have tested this and live-update is  working fine. 
Also live update has nothing to do with the modifier for this case and we actually rerender the whole table.

timabbott commented:
OK; I don't think this is worth merging independently, but can you stack the scrolling fix you'd like to do on top of this?

sahil839 commented:
Okay, so I should change the code in list_render to always consider simplebar as discussed [here](https://github.com/zulip/zulip/pull/15650#discussion_r451212578), correct?

sahil839 commented:
I have added the commits to fix the scrolling bug. I have changed all list_render instances to use simplebar. 

Also added a commit for rerendering the muted topics list only if settings overlay is opened and muted topics page is loaded.
Before adding this, if we were muting the topic from recent topics menu and we have not opened the muted topics yet, the simplebar container was not rendered and thus expectOne was returning error. It is better to rerender only when the overlay is open in the general case also.

amanagr commented on static/styles/settings.scss:
```diff
@@ -544,7 +544,7 @@ input[type=checkbox] {
     }
```
@sahil839 can you add `setting:` before the commit text and elaborate the commit a little.

sahil839 replied:
Added the explanation.

amanagr commented:
@sahil839 this LGTM, You can mark yourself as the author of the last commit if you would like to.

timabbott commented:
Merged, thanks @sahil839 and @amanagr!
